module/userProjects/helper.js

```diff
@@ -1069,10 +1069,13 @@ module.exports = class UserProjectsHelper {
 
             let userRoleInformation = _.omit(bodyData,["referenceFrom","submissions","hasAcceptedTAndC"]);
             if (projectId === "") {
+
+                const targetedSolutionId = await coreService.verifyTargetedSolution(userToken,bodyData,solutionId)
                 
                 const projectDetails = await projectQueries.projectDocument({
                     solutionId: solutionId,
-                    userId: userId
+                    userId: userId,
+                    isAPrivateProgram: targetedSolutionId.data.isATargetedSolution ? false : true
```

Comment:
@ankitpws Will private and public project have same solution id? I think it will be different, and if its different it this query might not work i think, just check through once, from app which solution id comes in targeted vs non targeted and based on that if this condition will work or not

Reply:
if public solution id used by non targeted user then this api will return solution is not targeted to user

Reply:
@aks30

Reply:
and the issue was only two project should be created once non targeted has project already in that case its a private project but when he turned in targeted then only solution which are targeted to him those are visible to him in that condition it will work

Reply:
I didn't understand we can connect over a call and discuss.